examples/chip-tool/commands/common/Command.cpp

```diff
@@ -54,6 +58,59 @@ bool Command::InitArguments(int argc, char ** argv)
     return isValidCommand;
 }
 
+bool Command::InitOptionalArguments(int & argc, char ** argv)
+{
+    bool isValidCommand = false;
+    int optArgsCount    = static_cast<int>(mOptionalArgs.size());
+    int argsCount       = static_cast<int>(mArgs.size());
+    int optionalArgC    = argc - argsCount;
+    int expectedArgs    = optionalArgC;
+    int optInits        = 0;
+
+    // If no optional args passed in, in return true.
+    if (optionalArgC == NO_OPTIONAL_ARGS)
+        return true;
+
+    // If too many optinoal args passed in, return false.
+    if (optArgsCount * OPTIONAL_ARGS_OFFSET < optionalArgC)
+    {
+        ChipLogError(chipTool, "Too Many arguments passed in.");
+        return false;
+    }
+    // Iterate the total configured optional args
+    for (int i = 0; i < optArgsCount; i++)
```

Comment:
So ideally optArgsCount and `i` would both be `size_t` here.  Any reason not to give them those types?